src/amo/components/OverallRating.js

```diff
@@ -1,37 +1,69 @@
 import React, { PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { compose } from 'redux';
 
+import { setUserRating } from 'amo/actions/ratingActions';
+import { callApi } from 'core/api';
 import translate from 'core/i18n/translate';
+import log from 'core/logger';
 
 import 'amo/css/OverallRating.scss';
 
 
 export class OverallRatingBase extends React.Component {
   static propTypes = {
-    i18n: PropTypes.object,
-    addonName: PropTypes.string,
+    addonName: PropTypes.string.isRequired,
+    addonID: PropTypes.number.isRequired,
+    authToken: PropTypes.string,
+    version: PropTypes.object.isRequired,
+    i18n: PropTypes.object.isRequired,
+    createRating: PropTypes.func.isRequired,
+  }
+
+  onChange = ({ currentTarget: form }) => {
+    const input = form.querySelector('input:checked');
+    if (!input) {
+      // I don't think any user input would cause a form change like
+      // this. I'm not sure what would cause this at all but it's probably
+      // better not to throw an exception.
+      log.error('The form changed but no input was selected');
```

Comment:
FWIW (since I'm not a screenreader user) trying the markup with fieldset, legend, labels and buttons  it reads out the following when focusing on the buttons.

* I love it, button, How do you feel about Ad Blocker?
* Huh?, button, How do you feel about Ad Blocker?
* I'm confused, button, How do you feel about Ad Blocker?